src/insert-extension.js

```diff
@@ -0,0 +1,116 @@
+ * when `insertAmpExtensionScript()` is called.
+ */
+let ampExtensionScriptInsertedOrPresent = Object.create(null);
+
+/**
+ * Reset the ampExtensionScriptInsertedOrPresent value for each test.
+ * @visibleForTesting
+ */
+export function resetExtensionScriptInsertedOrPresentForTesting() {
+  ampExtensionScriptInsertedOrPresent = Object.create(null);
+}
+
+/**
+ * Check script info in HTML head and make update if necessary
+ * @param {!Window} win
+ * @param {string} extension
+ */
+export function insertAmpExtensionScript(win, extension) {
+  if (extension == 'amp-embed') {
+    extension = 'amp-ad';
+  }
+  if (isAmpExtensionScriptRequired(win, extension)) {
+    const ampExtensionScript = createAmpExtensionScript(win, extension);
+    win.document.head.appendChild(ampExtensionScript);
+  }
+}
+
+/**
+ * Create the missing amp extension HTML script element.
+ * @param {!Window} win
+ * @param {string} extension
+ * @return {!HTMLScriptElement} Script object
+ */
+function createAmpExtensionScript(win, extension) {
+  const ampExtensionScript = win.document.createElement('script');
+  ampExtensionScript.async = true;
+  ampExtensionScript.setAttribute('custom-element', extension);
+  ampExtensionScript.setAttribute('data-script', extension);
+  const pathStr = win.location.pathname;
+  const useCompiledJs = getMode().test && window.ampTestRuntimeConfig &&
+      window.ampTestRuntimeConfig.useCompiledJs;
+  const scriptSrc = calculateExtensionScriptUrl(pathStr, extension,
+      getMode().test, useCompiledJs);
+  ampExtensionScript.src = scriptSrc;
+  return ampExtensionScript;
+};
+
+/**
+ * Determine the need to add amp extension script to document.
+ * @param {!Window} win
+ * @param {string} extension
+ * @return {boolean} Whether the action of adding an ampExtensionScript is required.
+ */
+function isAmpExtensionScriptRequired(win, extension) {
+  if (ampExtensionScriptInsertedOrPresent[extension]) {
+    return false;
+  }
+  const ampExtensionScriptInHead = win.document.head.querySelector(
+      `[custom-element="${extension}"]`);
+  ampExtensionScriptInsertedOrPresent[extension] = true;
+  return !ampExtensionScriptInHead;
+};
+
+/**
+ * Calculate script url for amp-ad.
+ * @visibleForTesting
+ * @param {string} path Location path of the window
+ * @param {string} extension
+ * @param {boolean=} isTest
+ * @param {boolean=} isUsingCompiledJs
+ * @return {string}
+ */
+export function calculateExtensionScriptUrl(path, extension, isTest,
+  isUsingCompiledJs) {
+  if (getMode().localDev) {
+    if (isTest) {
+      if (isUsingCompiledJs) {
+        return `/base/dist/v0/${extension}-0.1.js`;
+      }
+      return `/base/dist/v0/${extension}-0.1.max.js`;
+    }
+    if (path.indexOf('.max') >= 0) {
+      return `http://localhost:8000/dist/v0/${extension}-0.1.max.js`;
+    }
+    if (path.indexOf('.min') >= 0) {
+      return `http://localhost:8000/dist/v0/${extension}-0.1.js`;
```

Comment:
This entire file is gone in master. Were any changes made here? We could just remove the file in the PR if not.